### DISCUSSION TITLE
How to use React Server Components with Mantine AppShell?

### DISCUSSION THREAD

**Discussion by kotx** [mantinedev/mantine#5536] [Q&A]
There's a [demo](https://mantine.dev/app-shell/?e=BasicAppShell) where `useDisclosure` is used with an AppShell:
```ts
const [opened, { toggle }] = useDisclosure();

return (
  <AppShell
    header={{ height: 60 }}
    navbar={{
      width: 300,
      breakpoint: "sm",
      collapsed: { mobile: !opened },
    }}
    padding="md"
  >
    <AppShellHeader>
      <Group h="100%" px="md">
        <Burger opened={opened} onClick={toggle} hiddenFrom="sm" size="sm" />
        ...
      </Group>
    </AppShellHeader>
    ...
    <AppShellMain>{children}</AppShellMain>
);
  ```
But `useDisclosure` can't be used in server components.
Any way I can use AppShell with a collapsible menu without turning my entire layout into client components?

**Comment by pzehle** [comment]
Having the same question. I wouldn't mind an example or someone point me in the right direction. @kotx did you find anything?

**Reply by kotx** [reply]
Nope, I ended up moving off React altogether but I don't imagine that's viable for most.

**Comment by Kenzo-Wada** [comment]
@kotx 

As RSC cannot manage the state by itself, I think it is difficult to control the open state using a client function like useDisclosure with the current I/F of Appshell.

One possible work would be to define query params such as `?open=true` to control the open state using the server function!
